app/components/MixerVolmeter.vue.ts

```diff
@@ -118,21 +118,31 @@ export default class MixerVolmeter extends TsxComponent<MixerVolmeterProps> {
     this.gl = this.$refs.canvas.getContext('webgl', { alpha: false });
 
     if (this.gl) {
-      this.initWebglRendering();
-
-      // Get ready to lose this conext if too many are created
-      if (window['activeWebglContexts'] == null) window['activeWebglContexts'] = 0;
-      window['activeWebglContexts'] += 1;
-      this.$refs.canvas.addEventListener('webglcontextlost', this.handleLostWebglContext);
+      try {
+        this.initWebglRendering();
+
+        // Get ready to lose this conext if too many are created
+        if (window['activeWebglContexts'] == null) window['activeWebglContexts'] = 0;
+        window['activeWebglContexts'] += 1;
+        this.$refs.canvas.addEventListener('webglcontextlost', this.handleLostWebglContext);
+      } catch (e) {
+        console.error('Failed to initialize WebGL rendering, falling back to Canvas 2d', e);
+        this.gl = null;
+        this.initCanvas2dRendering();
+      }
```

Comment:
This code will be not actual after merging Volmeters on a shared canvas

Reply:
Good to know